o1vm/src/pickles/main.rs

```diff
@@ -65,11 +69,29 @@ pub fn main() -> ExitCode {
     // Initialize the environments
     let mut mips_wit_env =
         mips_witness::Env::<Fp, PreImageOracle>::create(cannon::PAGE_SIZE as usize, state, po);
-    let mut _mips_con_env = mips_constraints::Env::<Fp>::default();
+
+    // TODO: give this to the prover + verifier
+    let _constraints = {
+        let mut mips_con_env = mips_constraints::Env::<Fp>::default();
+        let mut constraints = Instruction::iter().fold(vec![], |mut acc, instr| {
```

Comment:
`iter` is not correctly used here. Detected while writing regression tests to check the number of constraitns and their degree. `IntoIter` must be reimplemented. Doing it.

Reply:
Or `flat_map` can be used. I think it is a better solution for now.